docs/source/app_developers_guide/overview.rst

```diff
@@ -0,0 +1,36 @@
+***************************
+Transaction Family Overview
+***************************
+
+Sawtooth separates the application level from the core system level with
+transaction families, which allows application developers to write in the
+languages of their choice. Each application defines the custom transaction
+families for its unique requirements.
+
+A transaction family includes these components:
+
+* A transaction processor to define the business logic for your application
+* A data model to record and store data
+* A client to handle the client logic for your application
+
+The transaction processor is responsible for registering with the validator,
+providing a transaction handler with an apply method and a transaction handler
+class for metadata, and getting/setting state as needed.
+
+The client is responsible for creating and signing transactions, combining
+those transactions into batches, and submitting them to the validator. The
+client can post batches through the REST API or connect directly to the
+validator via `ZeroMQ <http://zeromq.org>_.
```

Comment:
for the todo list.. we probably want to document this approach better somewhere. It's basically to create a zmq socket and then send the pre-defined protobufs (https://github.com/hyperledger/sawtooth-core/tree/master/protos). In order for a client to open the socket, a validator would expose the component port. That would I think in most cases be an internal facing connection in a DMZ. i.e. I don't think you want to expose the component port to the internet.